### PR TITLE
Advanced Fatigue - Improve misc. code

### PR DESCRIPTION
**Pull request by Salluci**
**When merged this pull request will:**
- Adds non-breaking adv fatigue changes from #8763.
- Add band-aid fix to non-fading stamina bar.
- Check ACE_Weather_Enabled variable for weather duty factor instead of checking if mod is loaded.
- Use hashMap for dutyFactor functions.
- Switch to LINKFUNC macro in player eventhandler.

### IMPORTANT

- [ ] If the contribution affects [the documentation](https://github.com/acemod/ACE3/tree/master/docs), please include your changes in this pull request so the documentation will appear on the [website](https://ace3mod.com/).
- [x] [Development Guidelines](https://ace3mod.com/wiki/development/) are read, understood and applied.
- [x] Title of this PR uses our standard template `Component - Add|Fix|Improve|Change|Make|Remove {changes}`.

### CODE REVIEW

File: addons/advanced_fatigue/XEH_postInit.sqf [hunk 1]
```diff
@@ -56,12 +56,12 @@ if (!hasInterface) exitWith {};
             [1, 3] select (_this getVariable [QEGVAR(dragging,isCarrying), false]);
         }] call FUNC(addDutyFactor);
     };
-    if (["ACE_Weather"] call EFUNC(common,isModLoaded)) then {
+        if (missionNamespace getVariable [QEGVAR(weather,enabled), false]) then { // Weather has an off switch, Dragging & Medical don't.
```
Comment (BaerMitUmlaut) [inline addons/advanced_fatigue/XEH_postInit.sqf:59]: Why the indentation? Also would prefer the comment in its own line.

File: addons/advanced_fatigue/XEH_postInit.sqf [hunk 2]
```diff
@@ -56,12 +56,12 @@ if (!hasInterface) exitWith {};
...
         [QEGVAR(weather,temperature), { // 35->1, 45->2
             linearConversion [35, 45, (missionNamespace getVariable [QEGVAR(weather,currentTemperature), 25]), 1, 2, true];
         }] call FUNC(addDutyFactor);
     };
 
     // - Add main loop at 1 second interval -------------------------------------------------------------
-    [FUNC(mainLoop), [], 1] call CBA_fnc_waitAndExecute;
+    [FUNC(mainLoop), 1] call CBA_fnc_addPerFrameHandler;
```
Comment (BaerMitUmlaut) [inline addons/advanced_fatigue/XEH_postInit.sqf:66]: No. `waitAndExecute` pauses when the game is paused, PFH doesn't.

File: addons/advanced_fatigue/functions/fnc_handleStaminaBar.sqf [hunk 3]
```diff
@@ -31,6 +31,8 @@ if (GVAR(fadeStaminaBar)) then {
     } else {
         _staminaBarContainer ctrlSetFade (0.9 * _stamina / 0.8);
     };
+} else {
+    _staminaBarContainer ctrlSetFade 0;
```
Comment (BaerMitUmlaut) [inline addons/advanced_fatigue/functions/fnc_handleStaminaBar.sqf:35]: Why?

**Review by BaerMitUmlaut — CHANGES_REQUESTED** [review]

File: addons/advanced_fatigue/functions/fnc_handleStaminaBar.sqf (hunk 3 above)
Comment (Salluci) [reply addons/advanced_fatigue/functions/fnc_handleStaminaBar.sqf:35]: With `fadeStaminaBar` set to false, ctrlFade isn't restored to 0 after death or initial respawn screen.

Comment (BaerMitUmlaut) [reply addons/advanced_fatigue/functions/fnc_handleStaminaBar.sqf:35]: Hmm I guess two more commands doesn't really hurt.

File: addons/advanced_fatigue/functions/fnc_mainLoop.sqf [hunk 4]
```diff
@@ -14,7 +14,10 @@
  *
  * Public: No
  */
-if (!alive ACE_player) exitWith { // Dead people don't breath, Will also handle null (Map intros)
+
+// Dead people don't breath, Will also handle null (Map intros)
```
Comment (BaerMitUmlaut) [inline addons/advanced_fatigue/functions/fnc_mainLoop.sqf:18]:
```suggestion
// Dead people don't breathe, will also handle null (map intros)
```

**Review by PabstMirror — APPROVED** [review]

**Review by veteran29 — APPROVED** [review]

**Pull request merged**